### PR TITLE
Feature/trace excluded paths

### PR DESCRIPTION
**Pull request by feniks1106**
- Separated middleware into 3 functions: GetMetricsMiddleware, GetTracingMiddleware, and GetLoggingMiddleware.
- Updated telemetry package and its usage
- Updated unit tests
- Marked Initialize() and Middleware() as depracated

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		}
+
+		for _, v := range paths {
+			opt.ExcludedPaths[strings.ToLower(v)] = struct{}{}
+		}
+	}
+}
+
+// MakeTracingOptions builds [TracingOptions] based on the provided options.
+func MakeTracingOptions(opts ...TracingOption) TracingOptions {
```
Comment (twistingmercury) [inline options.go:25]: Should probably be `New` instead of `Make`

File: options.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	opt := TracingOptions{}
+
+	for _, apply := range opts {
+		apply(&opt)
+	}
+
+	return opt
+}
+
+type MetricsOptions struct {
```
Comment (twistingmercury) [inline options.go:35]: let's comment these out. Don't delete, just comment.

**Review by twistingmercury — CHANGES_REQUESTED** [review]
A couple of small naming changes, and commenting out future code.

File: options.go (hunk 1 above)
Comment (feniks1106) [reply options.go]: Renamed to `NewTracingOptions`

**Review by twistingmercury — APPROVED** [review]
Merge it, my good man!

**Pull request merged**